File: lab/javascript/POO/contador_de_objetos.js
class Produzir {
  static contador = 0; // variável de classe (compartilhada)

  constructor() {
    // toda vez que um objeto é criado, incrementa
    Produzir.contador++;
  }

  toString() {
    return `Criados ${Produzir.contador} objetos`;
  }
}

// Teste
const p1 = new Produzir();
const p2 = new Produzir();
const p3 = new Produzir();

console.log(p3.toString());

File: lab/python/POO/contador_de_objetos.py
class Produzir:
    contador = 0  # variável de classe (compartilhada)

    def __init__(self):
        # toda vez que um novo objeto é criado:
        Produzir.contador += 1  # aumenta o contador da classe

    def __repr__(self):
        return f"Criados {self.contador} objetos"


p1 = Produzir()
p2 = Produzir()
p3 = Produzir()
print(p3)
